packages/webcert/src/store/api/apiMiddleware.ts

```diff
@@ -4,19 +4,25 @@ import { apiCallBegan, apiCallFailed, apiCallSuccess, ApiError, apiGenericError,
 import { AnyAction } from '@reduxjs/toolkit'
 import { throwError } from '../error/errorActions'
 import { createErrorRequestFromApiError, createSilentErrorRequestFromApiError } from '../error/errorCreator'
+import { FunctionBlocker, generateFunctionBlocker } from '../../components/utils/functionBlockerUtils'
 
 const handleApiCallBegan: Middleware = ({ dispatch }: MiddlewareAPI) => (next: Dispatch) => async (action: AnyAction) => {
   if (!apiCallBegan.match(action)) {
     return
   }
 
-  const { url, method, data, onStart, onSuccess, onError, onArgs } = action.payload
+  const { url, method, data, onStart, onSuccess, onError, onArgs, functionBlockerType } = action.payload
+  const functionBlocker: FunctionBlocker = generateFunctionBlocker()
```

Comment:
Only generate a functionBloocker if it is necessary. Now it will generate one even if there is no functionBlockerType.